### PR TITLE
CP-39128 & part of CP-39114: Add support for `CHCUpdates.xml` and `dev-CHCUpdates.xml`

### PR DESCRIPTION
**Pull request by danilo-delbusso**
To review commit-by-commit.

### CP-39128:

Adds `UPDATES_URL` as a placeholder value for the URL targeting the location of `CHCUpdates.xml`

### CP-39114:
Generates `CHCUpdates.xml` and `dev-CHCUpdates.xml`. Both use placeholders that are modified in the build pipeline: `UPDATES_URL` and `@DEV_MSI_URL_PLACEHOLDER@`

### CODE REVIEW

**Review by AaronRobson — APPROVED** [review]

**Comment by CitrixChris** [comment]
Just to clarify, youve created a placeholder for the url (where chcupdates is hosted) but not directly used any url in this PR?
Is the url passed in as pipeline param?

**Review by CitrixChris — APPROVED** [review]

**Comment by danilo-delbusso** [comment]
Blocking until the branding PR is ready.

File: scripts/xenadmin-build.sh [hunk 1]
```diff
@@ -217,7 +222,33 @@ cp ${REPO}/XenAdmin/bin/Release/{CommandLib.pdb,${BRANDING_BRAND_CONSOLE_NO_SPAC
 
 cd ${OUTPUT_DIR} && zip -r -m  ${BRANDING_BRAND_CONSOLE_NO_SPACE}.Symbols.zip *.pdb
 
-sha256sum ${OUTPUT_DIR}/${BRANDING_BRAND_CONSOLE_NO_SPACE}.msi > ${OUTPUT_DIR}/${BRANDING_BRAND_CONSOLE_NO_SPACE}.msi.checksum
+msi_checksum_with_file_name=`sha256sum ./CitrixHypervisorCenter.msi`
+echo $msi_checksum_with_file_name > ${OUTPUT_DIR}/${BRANDING_BRAND_CONSOLE_NO_SPACE}.msi.checksum
+msi_checksum=`echo $msi_checksum_with_file_name | cut -d " " -f 1`
```
Comment (kc284) [inline scripts/xenadmin-build.sh:227]: FYI, you don't need to change it, but you could also do
`msi_checksum=($msi_checksum_with_file_name)`

File: scripts/xenadmin-build.sh [hunk 2]
```diff
@@ -207,7 +212,7 @@ cd ${REPO}/XenAdmin/TestResources && zip -r ${OUTPUT_DIR}/${BRANDING_BRAND_CONSO
 cd ${REPO}/CFUValidator/bin/Release && zip ${OUTPUT_DIR}/CFUValidator.zip ./{*.dll,CFUValidator.exe,${BRANDING_BRAND_CONSOLE_NO_SPACE}Main.exe}
 
 #now package the pdbs
-cp ${REPO}/packages/*.pdb ${OUTPUT_DIR}
+[[ -e ${REPO}/packages/*.pdb ]] && cp ${REPO}/packages/*.pdb ${OUTPUT_DIR}
```
Comment (kc284) [inline scripts/xenadmin-build.sh:215]: Do we want this check here? I'd think we want it to fail if there are no pdbs so we know something went wrong.

File: Jenkinsfile [hunk 3]
```diff
@@ -31,7 +31,7 @@
  * SUCH DAMAGE.
  */
 
-def XENADMIN_BRANDING_TAG = 'v4.0'
+def XENADMIN_BRANDING_TAG = 'v4.3'
```
Comment (kc284) [inline Jenkinsfile:34]: As discussed, better use a branch for the time being. Same below for the pipeline.

**Review by kc284 — CHANGES_REQUESTED** [review]

File: scripts/xenadmin-build.sh (hunk 2 above)
Comment (danilo-delbusso) [reply scripts/xenadmin-build.sh:215]: It was failing for me, but the rest of the installation went ahead just fine. The pdbs for the packages were present in the Debug/Release subfolders in bin, but not in `packages`.

If that's a configuration mistake of my VS2019, I can remove this

File: scripts/xenadmin-build.sh (hunk 1 above)
Comment (danilo-delbusso) [reply scripts/xenadmin-build.sh:227]: `msi_checksum` is only for the checksum. The snippet removes the file's name that is returned from the `sha256` call.

```sh
PS C:\Users\danilod> sha256sum.exe .\test.txt
\e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 *.\\test.txt
```

I tried the following:
```sh
msi_checksum_with_file_name=`sha256sum ./test.sh`
msi_checksum=($msi_checksum_with_file_name)
echo $msi_checksum
```

But it returned this:
```sh
PS C:\Users\danilod> sh .\test.sh
)eba71bcf933510ba998149c20a4fbd9691f6c8573bc534d4c3cb517f2649783d *./test.sh
```

File: scripts/xenadmin-build.sh (hunk 2 above)
Comment (kc284) [reply scripts/xenadmin-build.sh:215]: Ah, just run the download_packages.ps1 adding the switch `-SYMBOLS`. (`Get-Help .\download_packages.ps1` shows the available switches).

File: scripts/xenadmin-build.sh (hunk 1 above)
Comment (danilo-delbusso) [reply scripts/xenadmin-build.sh:227]: EDIT. I was running this on ps. Your version works. I'll update the line

File: scripts/xenadmin-build.sh (hunk 2 above)
Comment (danilo-delbusso) [reply scripts/xenadmin-build.sh:215]: ok sounds good. I'll test it and if it works I'll remove the existence check 👍

**Comment by danilo-delbusso** [comment]
Branding/Pipeline PRs have been updated and merged. Unblocking this one. Only awaiting review of the latest changes.

**Comment by kc284** [comment]
Could we perhaps squash to 2 commits, one for each referenced ticket?

**Comment by danilo-delbusso** [comment]
> Could we perhaps squash to 2 commits, one for each referenced ticket?

All done 👍

**Comment by kc284** [comment]
Blocking this until the branding and pipeline corrections are merged.

**Pull request merged**